script.js

```diff
@@ -0,0 +1,53 @@
+const textToTypeContainer = document.querySelector('#textToTypeContainer')
+let referenceTextArray = []
+
+fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-generator&start_with_fixed=0&paragraphs=4').then((response) => {
+    return response.json()
+}).then((data) => {
+    data.forEach((textOutput) => {
+        //the line below replaces any accented character by the matching non-accented character
+        //e.g. "é" becomes "e"
+        textOutput = textOutput.normalize("NFD").replace(/\p{Diacritic}/gu, "")
+        wordsArray =  textOutput.split(' ')
+        referenceTextArray = referenceTextArray.concat(wordsArray)
+    }) 
+    
+    let count = 0
+    let stringForHtml = '<p class="textParagraph">'
+    referenceTextArray.forEach((word) => {
+        stringForHtml = stringForHtml + '<span id="' + count + '"> ' + word + ' </span>'
+        count++
+    })
+    stringForHtml += '</p>'
+    textToTypeContainer.innerHTML += stringForHtml
+})
+
+let wordIndexCount = -1
+let numberValidWords = 0
```

Comment:
we need to decide whether to use this, or a different way, as we have counted the red and green words in our code - doesn't matter either way, as both will work!

Also, `numberOfValidWords` or `validWordsCount`might be a little clearer